pkg/xds/generator/inbound_proxy_generator.go

```diff
@@ -42,7 +42,8 @@ func (g InboundProxyGenerator) Generate(ctx context.Context, _ *core_xds.Resourc
 			Configure(envoy_clusters.ProvidedEndpointCluster(false, core_xds.Endpoint{Target: endpoint.WorkloadIP, Port: endpoint.WorkloadPort})).
 			Configure(envoy_clusters.Timeout(defaults_mesh.DefaultInboundTimeout(), protocol))
 		// localhost traffic is routed dirrectly to the application, in case of other interface we are going to set source address to
-		if proxy.Dataplane.IsUsingTransparentProxy() && (endpoint.WorkloadIP != core_mesh.IPv4Loopback.String() || endpoint.WorkloadIP != core_mesh.IPv6Loopback.String()) {
+		// 127.0.0.6 to avoid redirections and thanks to first iptables rule just return fast
+		if proxy.Dataplane.IsUsingTransparentProxy() && (endpoint.WorkloadIP != core_mesh.IPv4Loopback.String() && endpoint.WorkloadIP != core_mesh.IPv6Loopback.String()) {
```

Comment:
@nicoche I think that's something you pointed out so would love a review :)

Reply:
tbh I don't fully grasp how this all is supposed to work; I'm not familiar with k8s setup nor with the default iptables, so take my review with a grain of salt.

In our case, `WorkloadIP` is a `10.0. ...` address. We would validate this new condition and Envoy's configuration would have:

```
"upstream_bind_config": {
  "source_address": {
    "address": "127.0.0.6",
    "port": 0,
  },
}
```

This does not work for us as trying to initiate a connection with a loopback source address and a non-loopback destination address seems illegal. Envoy requests fail and log `invalid argument`.

This is why I was thinking that the expected condition should be `if proxy.Dataplane.IsUsingTransparentProxy() && (endpoint.WorkloadIP == core_mesh.IPv4Loopback.String() || endpoint.WorkloadIP == core_mesh.IPv6Loopback.String()) {`

But maybe we're just doing something esoteric? 😕

Reply:
@bartsmykla to double check please

Reply:
It's correct